designer/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentService.scala

```diff
@@ -612,45 +632,39 @@ class DeploymentService(
   //       - send notifications about finished/failed custom actions
   private def processCustomAction(command: CustomActionCommand): Future[CustomActionResult] = {
     import command._
-    implicit val freshnessPolicy: DataFreshnessPolicy = DataFreshnessPolicy.Fresh
-    dbioRunner.run(
-      for {
-        // Fetch and validate process details
-        processDetailsOpt <- processRepository.fetchLatestProcessDetailsForProcessId[CanonicalProcess](
-          processIdWithName.id
-        )
-        processDetails <- existsOrFail(processDetailsOpt, ProcessNotFoundError(processIdWithName.name))
-        _ = checkIfCanPerformActionOnScenario(actionName, processDetails)
-        processState <- DBIOAction.from(getProcessState(processDetails))
-        manager = dispatcher.deploymentManagerUnsafe(processDetails.processingType)
-        // Fetch and validate custom action details
-        actionCommand = DMCustomActionCommand(
-          actionName,
-          processDetails.toEngineProcessVersion,
-          processDetails.json,
-          user.toManagerUser,
-          params
-        )
-        customActionOpt = manager.customActionsDefinitions.find(_.actionName == actionName)
-        customAction <- existsOrFail(
-          customActionOpt,
-          CustomActionNonExistingError(
-            s"Couldn't find definition of action ${actionName.value} for scenario ${processIdWithName.name} when trying to validate"
-          )
-        )
-        _ <- validateActionCommand(actionCommand, customAction)
-        _ = checkIfCanPerformActionInState(actionName, processDetails, processState)
-        invokeActionResult <- DBIOAction.from(manager.processCommand(actionCommand))
-      } yield invokeActionResult
-    )
+    for {
+      ctx <- prepareCommandContextWithAction[CanonicalProcess](
+        processIdWithName,
+        actionName,
+        params,
+        p => Some(p.processVersionId),
+        _ => None
+      )
+      dmCommand = DMCustomActionCommand(
+        actionName,
+        ctx.latestScenarioDetails.toEngineProcessVersion,
+        ctx.latestScenarioDetails.json,
+        user.toManagerUser,
+        params
+      )
+      actionResult <- runActionAndHandleResults(
+        actionName,
+        None,
```

Comment:
I'm considering adding `comment: Option[Comment]` to CustomActionCommand and use it here instead of this `None`. WDYT?